clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/ingestion/heartbeat/HeartbeatMonitoringService.java

```diff
@@ -339,6 +340,47 @@ public void updateLagMonitor(
     }
   }
 
+  /**
+   * Get maximum heartbeat lag from all regions for a given LEADER replica.
+   * @return Max leader heartbeat lag, or Long.MAX_VALUE if any region's heartbeat is unknown.
+   */
+  public long getReplicaLeaderHeartbeatLag(String storeName, int version, int partitionId) {
+    Map<String, HeartbeatTimeStampEntry> replicaTimestampMap =
+        getLeaderHeartbeatTimeStamps().getOrDefault(storeName, Collections.emptyMap())
+            .getOrDefault(version, Collections.emptyMap())
+            .getOrDefault(partitionId, Collections.emptyMap());
+    if (replicaTimestampMap.isEmpty()) {
+      return Long.MAX_VALUE;
+    }
+    long minTimestamp = Long.MAX_VALUE;
+    for (Map.Entry<String, HeartbeatTimeStampEntry> entry: replicaTimestampMap.entrySet()) {
+      if (!entry.getValue().consumedFromUpstream) {
+        return Long.MAX_VALUE;
+      }
+      minTimestamp = Math.min(minTimestamp, entry.getValue().timestamp);
```

Comment:
This.... will return the MIN lag from different regions right?  The javadoc says this computes the max lag?

Reply:
I think the method returns (System.current() - min valid HB TS) should be max HB lag

Reply:
Updated in the new commit - since we need to support lag logging, I changed to the way you suggested.